GetBrokerRedirectURI.ps1

```diff
@@ -0,0 +1,103 @@
+<#
+.SYNOPSIS
+Obtain the RedirectURI from an APK.  Requires JDK\bin\keytool and AndroidSDK\build-tools\###\aapt to be on the path.
+
+.EXAMPLE
+GetBrokerRedirectURI MyApp.apk
+#>
+Param(
+[Parameter(Mandatory=$true)] 
+[string]$apkFile
+)
+
+<#
+Extract all of the signature owner and SHA1 hex bytes from `keytool -printcert -jarfile`.
+#>
+function GetSHA1Hex {
+    Param(
+        [string]$apkFile
+        )
+    $output = keytool -printcert -jarfile $apkFile
+    $ownerPrefix = "Owner:"
+    $shaPrefix = "SHA1:"
+    $signatures = @{}
+    $owner = ""
+    ForEach ($line in $output) {
+        if ($line.contains($ownerPrefix)) {
+            $owner = $line.replace($ownerPrefix, "").trim()
+        }
+
+        if ($line.contains($shaPrefix)) {
+            $base64 = GetBase64FromSHA1Hex($line.replace($shaPrefix, "").trim())
+            $signatures.add($owner, $base64)
```

Comment:
this assumes that the owner line always comes before the sha1 line -- is that always true, even when there are multiples?

Reply:
Yup - I confirmed with an APK that has this scenario.  The owner always appears before the signatures.